### PR TITLE
Fetch and use the canonical entity ID when importing bundles

### LINKED ISSUE #2021: Once a multi-series subordinate is deployed, no relations can be made with it
Add "filebeat" to the canvas in any mode (sandbox, gijoe, etc) and then deploy it.
Add "elasticsearch" to the canvas and try and relate them. 
You won't be able to even though there is a possible relation.

If you add both to the canvas before deploying you will be able to create this relation

### PR DESCRIPTION
**Pull request by hatched**
It's possible that a user supplied charm ID is not the canonical ID. This would mean an ID mismatch between the charm that was added and then the application that was deployed causing errors on deploy.

Fixes #2021

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

Deploy this version of the GUI to AWS and then deploy the Observable-Kubernetes bundle. If it doesn't fail when hitting commit then it's good!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5849/
Test PASSed.

**Comment by urosj** [comment]
On QA now.

File: jujugui/static/gui/src/app/jujulib/charmstore.js [hunk 1]
```diff
@@ -261,6 +261,32 @@ var module = module;
...
+          return;
+        }
+        callback(null, data.Id);
+      };
+      jujulib._makeRequest(
+        this.bakery,
+        this._generatePath(entityId.replace('cs:', ''), null, '/meta/id'),
+        'GET',
+        null,
+        handler.bind(this));
```
Comment (frankban) [inline jujugui/static/gui/src/app/jujulib/charmstore.js]: Is the call to bind required here?

File: jujugui/static/gui/src/app/jujulib/test-charmstore.js [hunk 2]
```diff
@@ -498,6 +498,38 @@ describe('jujulib charmstore', function() {
...
   });
 
+  describe('getCanonicalId', function() {
+    it('makes a request to fetch the canonical id for an entity', function() {
+      const callback = sinon.stub();
+      charmstore.getCanonicalId('cs:xenial/ghost-4', callback);
+      const sendGetRequest = charmstore.bakery.sendGetRequest;
+      assert.equal(sendGetRequest.callCount, 1);
+      assert.equal(
+        sendGetRequest.args[0][0], 'local/v5/xenial/ghost-4/meta/id');
```
Comment (frankban) [inline jujugui/static/gui/src/app/jujulib/test-charmstore.js]: The following part of the test would be easier to read if we assigned sendGetRequest.args[0][0] and sendGetRequest.args[0][1] to variables, and then assert on those variables.

File: jujugui/static/gui/src/app/jujulib/test-charmstore.js [hunk 3]
```diff
@@ -498,6 +498,38 @@ describe('jujulib charmstore', function() {
...
+      // Call the success request callback
```
Comment (frankban) [inline jujugui/static/gui/src/app/jujulib/test-charmstore.js]: Final dot to the sentence.

**Comment by urosj** [comment]
QA 🆗 (seeing some unrelated GCE issues though)

File: jujugui/static/gui/src/app/jujulib/charmstore.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/jujulib/charmstore.js]: Nope it isn't, good catch.

File: jujugui/static/gui/src/app/jujulib/test-charmstore.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/jujulib/test-charmstore.js]: updated

File: jujugui/static/gui/src/app/jujulib/test-charmstore.js (hunk 3 above)
Comment (hatched) [inline jujugui/static/gui/src/app/jujulib/test-charmstore.js]: done

**Comment by hatched** [comment]
Thanks all!

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5850/
Test PASSed.

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1913

**Comment by hatched** [comment]
Spurious

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**